ledger/metrics/src/main/scala/com/daml/telemetry/TelemetryContext.scala

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.telemetry
+
+import java.util.{HashMap => jHashMap, Map => jMap}
+
+import com.daml.dec.DirectExecutionContext
+import io.grpc.MethodDescriptor
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Context
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+trait TelemetryContext {
+
+  /** Sets or replaces the value of `attribute` to `value`.
+    */
+  def setAttribute(attribute: SpanAttribute, value: String): TelemetryContext
```

Comment:
These telemetry classes are moved almost one-to-one from the oem-integration-kit. I just changed the API to use `SpanAttribute` instead of `AttributeKey` from `opentelemetry`.

Reply:
I'd prefer if we kept this as-is, i.e., let's do simplifications and/or changes to the `TelemetryContext` in a separate PR.

Reply:
Restored the changes. I also noticed 2 more issues with the `TelemetryContext` which I'm going to mention here and address in the next PR.